monai/data/image_writer.py

```diff
@@ -0,0 +1,395 @@
+        contiguous: bool = False,
+    ):
+        """
+        Rearrange the data array axes to make the `channel_dim`-th dim the last
+        dimension and ensure there are three spatial dimensions. If
+        ``channel_dim`` is ``None``, a new axis will be appended to the last
+        dimension.
+
+        When ``squeeze_end_dims`` is ``True``, a postprocessing step will be
+        applied to remove any trailing singleton dimensions.
+
+        Args:
+            data: input data to be converted to "channel-last" format.
+            channel_dim: specifies the axis of the data array that is the channel dimension.
+                ``None`` indicates no channel dimension.
+            squeeze_end_dims: if ``True``, any trailing singleton dimensions will be removed (after the channel
+                has been moved to the end). So if input is `(H,W,D,C)` and C==1, then it will be saved as `(H,W,D)`.
+                If D is also 1, it will be saved as `(H,W)`. If ``False``, image will always be saved as `(H,W,D,C)`.
+            spatial_ndim: modifying the spatial dims if needed, so that output to have at least
+                this number of spatial dims. If ``None``, the output will have the same number of
+                spatial dimensions as the input.
+            contiguous: if ``True``, the output will be contiguous.
+        """
+        # change data to "channel last" format
+        if channel_dim is not None:
+            # _chns = ensure_tuple_rep(channel_dim)
+            # data = moveaxis(data, _chns, tuple(range(-len(_chns), 0)))
+            data = moveaxis(data, channel_dim, -1)
+        else:  # adds a channel dimension
+            data = data[..., None]
+        # To ensure at least three spatial dims
+        if spatial_ndim:
+            while len(data.shape) < spatial_ndim + 1:  # assuming the data has spatial + channel dims
+                data = data[..., None, :]
+            while len(data.shape) > spatial_ndim + 1:
+                data = data[..., 0, :]
+        # if desired, remove trailing singleton dimensions
+        while squeeze_end_dims and data.shape[-1] == 1:
+            data = np.squeeze(data, -1)
+        if contiguous:
+            data = ascontiguousarray(data)
+        return data
+
+    @classmethod
+    def get_meta_info(cls, metadata: Optional[Mapping] = None):
+        """
+        Extracts relevant meta information from the metadata object (using ``.get``).
+
+        This method returns the following fields (the default value is ``None``):
+
+            - ``'original_affine'``: for data original affine (before any image processing),
+            - ``'affine'``: for the current data affine (representing the current coordinate information),
+            - ``'spatial_shape'``: for data original spatial shape.
+        """
+        if not metadata:
+            default_dict = {"original_affine": None, "affine": None, "spatial_shape": None}
+            metadata = default_dict
```

Comment:
Can be simplified: `metadata = {"original_affine": None, "affine": None, "spatial_shape": None}`
And seems no need to set default with `None`, just `metadata = {}` is same?

Thanks.